apps/verification/src/verification.service.ts

```diff
@@ -187,17 +187,15 @@ export class VerificationService {
       return getProofPresentationById?.response;
     } catch (error) {
       this.logger.error(`[getProofPresentationById] - error in get proof presentation by proofId : ${JSON.stringify(error)}`);
-      const errorStack = error?.response?.error?.reason;
+      const errorMessage = error?.response?.error?.reason;
 
-      if (errorStack) {
-        throw new RpcException({
-          message: ResponseMessages.verification.error.proofNotFound,
-          statusCode: error?.response?.status,
-          error: errorStack
-        });
-      } else {
-        throw new RpcException(error.response ? error.response : error);
+      // Handle "record not found" error with 404
+      if (errorMessage?.includes('record with id') && errorMessage?.includes('not found')) {
```

Comment:
I suppose we would also need to handle errors with `error?.response?.error?.reason` which might have error others than we have added conditions for which might not be directly available under `err.response`

Reply:
Hi! Thanks a lot for the feedback.
I’ve updated the logic based on your notes. Here's what I’m currently doing:

1. **If the error clearly indicates a "record not found"**, I return a `NotFoundException`:
```ts
if (errorMessage?.includes('record with id') && errorMessage?.includes('not found')) {
  throw new NotFoundException({
    statusCode: 404,
    message: errorMessage,
    error: 'Not found',
  });
}
```

2. **If the error has a structured shape**, I return it as an `RpcException`:
```ts
if (error?.response) {
  throw new RpcException({
    statusCode: error.response.status || 500,
    message: errorMessage || 'Internal server error',
    error: error.response.error || 'Internal error'
  });
}
```

3. **If the error doesn't follow a known structure**, I fall back to a generic `RpcException`:
```ts
throw new RpcException({
  statusCode: 500,
  message: 'Something went wrong!',
  error: errorMessage || 'Unknown error'
});
```
```suggestion
      if (errorMessage?.includes('record with id') && errorMessage?.includes('not found')) {
```
Let me know if this direction looks better or if you'd like me to change how the fallback is handled.

Reply:
This seems almost right, however, in either of the conditions, we don't want to ignore messages from error if it has `error.response` but doesn't have error.response.status.
Similar, to the previous implementation:
```ts
throw new RpcException(error.response ? error.response : error);
```

Reply:
I've updated the error handling logic based on your note. Here's what I'm doing now:

- I'm extracting the `errorMessage` like this:
```ts
const errorMessage = error?.response?.error?.reason || error?.message;
```
This ensures we capture a meaningful message even if `error.response` exists but doesn't include a structured `error.reason`.

- If the message indicates "record not found", I return a `NotFoundException`:
```ts
if (errorMessage?.includes('record with id') && errorMessage?.includes('not found')) {
  throw new NotFoundException({
    statusCode: 404,
    message: errorMessage,
    error: 'Not found'
  });
}
```

- For all other cases, I preserve the original structure:
```ts
throw new RpcException(error.response ? error.response : error);
```

Let me know if this looks aligned with what you had in mind, or if you'd like any further adjustments

Reply:
Okay this seems alright to me, we can handle it that way

Reply:
Hi @GHkrishna and @bhavanakarwade ,

I’ve committed the changes with the updated error handling. Let me know if anything else is needed!

Thanks!